temba/contacts/models.py

```diff
@@ -371,6 +377,11 @@ def set_field(self, user, key, value, label=None):
             else:
                 loc_value = self.org.parse_location(value, 1)
 
+            if loc_value is not None and len(loc_value) > 0:
+                loc_value = loc_value.first()
+            else:
+                loc_value = None
+
```

Comment:
No, I changed the implementation to return a list since we want to query admin boundaries with out going through higher admin levels. that way we can decide on what to do more than one results is returned (example  tofa ward exists in Kano and Sokoto)